### ISSUE TITLE
No Audio After 2023-03-08 Updates POP OS 22.04 Kernel Update 6.2

### ISSUE THREAD

**Issue by ganakee** [pop-os/pop#2826]
**Summary**
After a standard update including kernel 6.2 on 2023.03.08, all audio lost--no audio devices recognized. Cannot update and keep audio.

**Distribution (run `cat /etc/os-release`):**
 
```
 cat /etc/os-release
NAME="Pop!_OS"
VERSION="22.04 LTS"
ID=pop
ID_LIKE="ubuntu debian"
PRETTY_NAME="Pop!_OS 22.04 LTS"
VERSION_ID="22.04"
HOME_URL="https://pop.system76.com"
SUPPORT_URL="https://support.system76.com"
BUG_REPORT_URL="https://github.com/pop-os/pop/issues"
PRIVACY_POLICY_URL="https://system76.com/privacy"
VERSION_CODENAME=jammy
UBUNTU_CODENAME=jammy
LOGO=distributor-logo-pop-os
```

 
**Related Application and/or Package Version (run `apt policy $PACKAGE NAME`):**
Pipewire and audio (NOTE: this is the RESTORED version not possibly the problem version which would be the version exactly after this version, if any).
```
sudo apt policy pipewire
pipewire:
  Installed: 0.3.66~1676588331~22.04~135fd49
  Candidate: 0.3.66~1676588331~22.04~135fd49
  Version table:
 *** 0.3.66~1676588331~22.04~135fd49 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
        100 /var/lib/dpkg/status
     0.3.48-1ubuntu3 500
        500 http://apt.pop-os.org/ubuntu jammy-updates/main amd64 Packages
     0.3.48-1ubuntu1 500
        500 http://apt.pop-os.org/ubuntu jammy/main amd64 Packages
```

Hardware summary
Intel i7-4700
NVIDIA GTX760 (max NVIDIA driver 470)
Memory 32.0GB
Audio devices attached --headphones rear jack.

**Issue/Bug Description:**
SEE OTHER NOTES for details.
After a standard daily update on 2023.03.08, after reboot, all audio was lost.

1.  AUDIO icon missing from the TOP BAR,
2. opening Settings...Sound showed no audio devices available
3. running sudo systemctl --user restart pipewire
Failed to connect to bus: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user) 

**Steps to reproduce (if you know):**
Run updates via POP!_Shop (2023.03.08).
Reboot.
All audio lost.

**Expected behavior:**
Normal audio.

**Other Notes:**
I tried reinstalling pipewire without success. `sudo apt reinstall pipewire pipewire-pulse`

I needed to run a Timeshift restore to get working audio again as this is my main machine. After the Timeshift restore, I ran an apt update to see what recent updates occurred. Possible sources of the problem (diff between restored backup from 2023.03.07 and problem update):
```
sudo apt list --upgradeable
Listing... Done
gnome-online-accounts/jammy 3.44.0-1ubuntu1pop1~1677167546~22.04~5d98aa0 amd64 [upgradable from: 3.40.1-2ubuntu1pop0~1649081188~22.04~8042754]
libegl-mesa0/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
libgbm1/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
libgl1-mesa-dri/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
libgl1-mesa-dri/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 i386 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
libglapi-mesa/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
libglapi-mesa/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 i386 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
libglx-mesa0/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
libglx-mesa0/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 i386 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
libgoa-1.0-0b/jammy 3.44.0-1ubuntu1pop1~1677167546~22.04~5d98aa0 amd64 [upgradable from: 3.40.1-2ubuntu1pop0~1649081188~22.04~8042754]
libgoa-1.0-common/jammy,jammy 3.44.0-1ubuntu1pop1~1677167546~22.04~5d98aa0 all [upgradable from: 3.40.1-2ubuntu1pop0~1649081188~22.04~8042754]
libgoa-backend-1.0-1/jammy 3.44.0-1ubuntu1pop1~1677167546~22.04~5d98aa0 amd64 [upgradable from: 3.40.1-2ubuntu1pop0~1649081188~22.04~8042754]
libgranite-common/jammy,jammy 6.2.0pop1~1677598542~22.04~66414db all [upgradable from: 6.0.0pop0~1646751002~22.04~657345a]
libgranite6/jammy 6.2.0pop1~1677598542~22.04~66414db amd64 [upgradable from: 6.0.0pop0~1646751002~22.04~657345a]
libpop-system-updater-gtk/jammy 0.1.0~1678118957~22.04~2e50214 amd64 [upgradable from: 0.1.0~1677528343~22.04~740d3ab]
libtiff5/jammy-security,jammy-updates 4.3.0-6ubuntu0.4 amd64 [upgradable from: 4.3.0-6ubuntu0.3]
libxatracker2/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
linux-firmware/jammy,jammy 20230213.gitbf4115c1-0ubuntu1+system76~1677174484~22.04~76a1c5b all [upgradable from: 20230120.gitbb2d42dc-0ubuntu1+system76~1675385032~22.04~7cff33b]
linux-generic/jammy 6.2.0.76060200.202302191831~1677858327~22.04~3cea1be amd64 [upgradable from: 6.1.11.76060111.202302091138~1675975749~22.04~f771a7f]
linux-headers-generic/jammy 6.2.0.76060200.202302191831~1677858327~22.04~3cea1be amd64 [upgradable from: 6.1.11.76060111.202302091138~1675975749~22.04~f771a7f]
linux-image-generic/jammy 6.2.0.76060200.202302191831~1677858327~22.04~3cea1be amd64 [upgradable from: 6.1.11.76060111.202302091138~1675975749~22.04~f771a7f]
linux-system76/jammy 6.2.0.76060200.202302191831~1677858327~22.04~3cea1be amd64 [upgradable from: 6.1.11.76060111.202302091138~1675975749~22.04~f771a7f]
mesa-va-drivers/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
mesa-vdpau-drivers/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
mesa-vulkan-drivers/jammy 22.3.5-1pop1~1677168864~22.04~8f20e98 amd64 [upgradable from: 22.3.4-1pop1~1676988728~22.04~8cd56c0]
pop-cosmic/jammy,jammy 0.1.0~1677788568~22.04~c1895ce all [upgradable from: 0.1.0~1675712569~22.04~2ea0e71]
pop-system-updater/jammy 0.1.0~1678118957~22.04~2e50214 amd64 [upgradable from: 0.1.0~1677528343~22.04~740d3ab]
rsync/jammy-security,jammy-updates 3.2.7-0ubuntu0.22.04.2 amd64 [upgradable from: 3.2.3-8ubuntu3.1]
system76-acpi-dkms/jammy 1.0.2~1678132042~22.04~ed01124 amd64 [upgradable from: 1.0.2~1659568831~22.04~a88dcce]
systemd-hwe-hwdb/jammy-updates,jammy-updates 249.11.3 all [upgradable from: 249.11.2]
xserver-xorg-video-amdgpu/jammy 22.0.0-1ubuntu0.1pop1~1677198820~22.04~2e82f80 amd64 [upgradable from: 22.0.0-1ubuntu0.1]
```

**Comment by ganakee** [comment]
I had some more testing time today.

The issue seems narrowed to the 6.2 kernel update released on or about 2023.03.08 (`6.2.0.76060200.202302191831~1677858327~22.04`).

I marked all the kernel header packages for HOLD. (This tells APT not to install these packages.)

`sudo apt-mark hold linux-system76 linux-libc-dev linux-image-generic linux-headers-generic linux-generic`

One can verify the APT hold with
```
sudo apt-mark showhold
   linux-generic
   linux-headers-generic
   linux-image-generic
   linux-libc-dev
   linux-system76
   nvidia-driver-510
```
(To the keen eyed, I previously marked the 510 NVidia package as HOLD to prevent support problems with my graphics card GTX-760.)

I then installed all the other updates from 2023-03-08 to 2023-03.10. 

I re-booted and audio/sound works fine.

Thus, the issue seems to be the newly released Kernel 6.2 (`6.2.0.76060200.202302191831~1677858327~22.04`).

**Comment by shaunie2fly** [comment]
After the same kernel update my machine would start up.   Looks like some file system error.  Rolled back to the previous kernel version and working again as per usual.

**Comment by shaunie2fly** [comment]
**6.2.0-76060200-generic** - this is the naughty Kernel version.
**6.1.11-76060111-generic** - rolled back to this guy.

rolled back like this : 
`sudo kernelstub -v -k /boot/vmlinuz-6.1.11-76060111-generic -i /boot/initrd.img-6.1.11-76060111-generic`

**Comment by ganakee** [comment]
POP OS released a revised 6.2 Kernel on or about 2023-03-24. The revised 6.2 kernel seems to fix the no audio issue on POP OS 22.04 after the earlier,  Kernel 6.2 update 6.2.0-76060200-generic that I encountered.

The revised 6.2 Kernel and working kernel is 6.2.6-76060206-generic.
```
uname -a
Linux pop-os 6.2.6-76060206-generic #202303130630~1679424972~22.04~4a8cde1 SMP PREEMPT_DYNAMIC Tue M x86_64 x86_64 x86_64 GNU/Linux
```

The package of updates during the week of 2023-03-24 also apparently updated Pipewire.
to 0.3.67~1678985654~22.04~e2b5db7:
```
sudo apt policy pipewire
  Installed: 0.3.67~1678985654~22.04~e2b5db7
  Candidate: 0.3.67~1678985654~22.04~e2b5db7
  Version table:
 *** 0.3.67~1678985654~22.04~e2b5db7 1001
       1001 http://apt.pop-os.org/release jammy/main amd64 Packages
        100 /var/lib/dpkg/status
     0.3.48-1ubuntu3 500
        500 http://apt.pop-os.org/ubuntu jammy-updates/main amd64 Packages
     0.3.48-1ubuntu1 500
        500 http://apt.pop-os.org/ubuntu jammy/main amd64 Packages
```

If you followed @my instructions above to HOLD the problem packages,

1. do a manual backup before trying the corrected kernel
2. open terminal and un-hold the packages using `sudo apt-mark unhold linux-system76 linux-libc-dev linux-image-generic linux-headers-generic linux-generic`
3. run `sudo apt update`
4. run `sudo apt -y dist-upgrade` (auto-installs all updated packages, leave out -y if you wish to inspect first)
5. the updates may take a while due to building the kernel and installing new kernel images
6. reboot
7. open settings and sound to test audio
Closing on 2023-03-26.
